### DISCUSSION TITLE
Standalone widgets seem to always align to the right?

### DISCUSSION THREAD

**Discussion by mark8044** [daohoangson/flutter_widget_from_html#725] [Q&A] [answered]
Using your previous advice, I made a link widget to give me a stylized custom appearance to links which is overall working well.

```
          meta.register(BuildOp(
            onTree: (meta, tree) {
              
              WidgetBit.inline(tree.parent!, _LinkWidget(text:meta.element.text, url:meta.element.attributes['href'].toString())).insertBefore(tree);
              tree.detach();

              },
          ));
```

This is inside the `parse` method
```
void parse(BuildMetadata meta) { ... }
```

They render perfectly when they are within text, but if they are by themself and there is no free text in the Htmlwidget, only this rendered link widget, they for some reason are justified to the far right of the container.  Is there a simple/easy way to have them justify/align to the left?

**Comment by daohoangson** [comment, marked as answer]
This looks right. What is your HTML?

**Reply by mark8044** [reply]
This happens for both my custom SmilieWidget and LinkWidget, here's an example of a smilie:

```
<img src="/thumbsup.gif" class="smiley" />
```

Thats the entire code being given to HtmlWidget.

For now I've wrapped HtmlWidget in an Align tag, and made the alignment left and that seems to align everything leftward, but it seems like for some reason everything is aligned right.  

Maybe this is my fault and somewhere higher up in the widget tree I've set things to align right

---
May I ask a separate question:  With your help in the past, I was able to give an intrinsic padding to all my text using a textPadding widget, that simply wraps the widget in a padding element as follows:

```
class _TextPadding extends StatelessWidget {
  final Widget child;
  final double padding;

  const _TextPadding(this.child, this.padding, {Key? key}) : super(key: key);

  @override
  Widget build(BuildContext _) =>
      Padding(
          child: child,
          padding: EdgeInsets.symmetric(horizontal: padding));
}
```

Using this, I can give padding to my text as follows:
```
  Widget? buildText(BuildMetadata meta, TextStyleHtml tsh, InlineSpan text) {
    var built = super.buildText(meta, tsh, text);

    if (built != null) {

            built = _TextPadding(built, textPadding);

```

We did this because I didn't want images to carry any padding, so those would be outside of the `buildText` and successfully have no padding.

My Smilie or Link widgets also get the padding when they are surrounded by text.

However if a smilie or Link widget it the only thing being rendered by `HtmlWidget` then there is no padding and it looks unnatural.

Is there a way to wrap the smilie or link widgets with that `_TextPadding` widget when its the only thing being rendered by `HtmlWidget` ?

I was able to successfully do it with another type of quote widget like this:
```
      meta.register(BuildOp(
        onWidgets: (meta, widgets) => listOrNull(
          buildColumnPlaceholder(meta, widgets)?.wrapWith(
                (_, child) {

                  if (meta.element.parent == null)
                    {
                      return _TextPadding(_QuoteWidget(child: child), textPadding);   // first quote
                    }
                  else
                    {
                      return _QuoteWidget(child: child);    //nested quotes (no text padding)
                    }

                }
          ),
        ),
      ));
```

But the way the smilies are being rendered I cannot figure out how to determine if its the only thing being rendered and should therefor carry padding:

```
 Widget buildImage(BuildMetadata meta, ImageMetadata data) {

    if (meta.element.className == 'smiley')
      {
        return Image.asset(_smilieName);
      }
```
